internal/driverutil/operation.go

```diff
@@ -30,3 +37,35 @@ const (
 	UpdateOp            = "update"            // UpdateOp is the name for updating
 	BulkWriteOp         = "bulkWrite"         // BulkWriteOp is the name for client-level bulk write
 )
+
+func CalculateMaxTimeMS(ctx context.Context, rttMin time.Duration, rttStats string, err error) (int64, error) {
```

Comment:
It's unconventional to pass the error to be returned in as a parameter. Instead, consider letting `CalculateMaxTimeMS` return just the `int64` value, and let the caller form the error value.

E.g. use in `batch_cursor.go`:
```go
maxTimeMS = driverutil.CalculateMaxTimeMS(ctx, rttMonitor.Min())
if maxTimeMS <= 0 {
	return nil, fmt.Errorf(
		"calculated server-side timeout (%v ms) is less than or equal to 0 (%v): %w",
		maxTimeMS,
		rttMonitor.Stats(),
		ErrDeadlineWouldBeExceeded)
}
```